docs/intro/overview.rst

```diff
@@ -13,172 +13,89 @@ precisely, `web scraping`_), it can also be used to extract data using APIs
-
-   ...
-
-.. highlight:: none
 
-An XPath expression to select the description could be::
+    class StackOverflowSpider(scrapy.Spider):
+        name = 'stackoverflow'
+        start_urls = ['http://stackoverflow.com/questions?sort=votes']
 
-    //div[@id='description']
+        def parse(self, response):
+            for href in response.css('.question-summary h3 a::attr(href)'):
+                full_url = response.urljoin(href.extract())
+                yield scrapy.Request(full_url, callback=self.parse_question)
 
-.. highlight:: html
+        def parse_question(self, response):
+            title = response.css('h1 a::text').extract_first()
+            votes = response.css('.question .vote-count-post::text').extract_first()
+            tags = response.css('.question .post-tag::text').extract()
+            body = response.css('.question .post-text').extract_first()
+            yield {
+                'title': title,
+                'votes': votes,
+                'body': body,
+                'tags': tags,
+                'link': response.url,
+            }
 
-Finally, the file size is contained in the second ``<p>`` tag inside the ``<div>``
-tag with ``id=specifications``::
 
-   <div id="specifications">
+Put this in a file, name it to something like ``stackoverflow_spider.py``
+and run the spider using the :command:`runspider` command::
 
-   <p>
-   <strong>Category:</strong>
-   <a href="/cat/4">Movies</a> &gt; <a href="/sub/35">Documentary</a>
-   </p>
+    scrapy runspider stackoverflow_spider.py -o top-stackoverflow-questions.json
 
-   <p>
-   <strong>Total size:</strong>
-   150.62&nbsp;megabyte</p>
 
+When this finishes you will have in the ``top-stackoverflow-questions.json`` file
+a list of the most upvoted questions in StackOverflow in JSON format, containing the
+title, link, number of upvotes, a list of the tags and the question content in HTML.
 
-.. highlight:: none
 
-An XPath expression to select the file size could be::
+What just happened?
+-------------------
 
-   //div[@id='specifications']/p[2]/text()[2]
+When you ran the command ``scrapy runspider somefile.py``, Scrapy looked
+for a Spider definition inside it and ran it through its crawler engine.
 
-.. highlight:: python
+The crawl started by making requests to the URLs defined in the ``start_urls``
+attribute (in this case, only the URL for StackOverflow top questions page),
+and then called the default callback method ``parse`` passing the response
+object as an argument.
 
-For more information about XPath see the `XPath reference`_.
+Here you notice one of the main advantages about Scrapy: requests are
+scheduled and processed asynchronously. This means that Scrapy doesn't
+need to wait for a request to be finished and processed, it can send
+another request or do other things in the meantime, which results in much
+faster crawlings.
```

Comment:
There is a balance between speed and politeness: users can send multiple requests in parallel about as easy with e.g. concurrent.futures + requests, but this way politeness settings (download delay, concurrency) won't be respected. The advantage of Scrapy is that it provides helpers to maintain per-domain politeness settings and still send async requests when possible (+ other benefits like automatic retries). This all could be much harder to implement with e.g. requests + a thread pool.